main/pipeline/semantic_extension/SemanticExtension.h

```diff
@@ -28,6 +28,7 @@ namespace pipeline::semantic_extension {
 class SemanticExtension {
 public:
     virtual void finishTypecheckFile(const core::GlobalState &, const core::FileRef &) const = 0;
+    virtual void finishTypecheckProject(const core::GlobalState &) const = 0;
```

Comment:
do we use the word Project anywhere? Maybe drop `Project` and just `finishTypecheck`?